Apps/Gateway/src/ClientApp/src/stores/user.ts

```diff
@@ -0,0 +1,468 @@
+                .then((userProfile) => {
+                    setProfileUserData(userProfile);
+                    resolve();
+                })
+                .catch((resultError: ResultError) => {
+                    setUserError(resultError.resultMessage);
+                    reject(resultError);
+                });
+        });
+    }
+
+    function recoverUserAccount(): Promise<void> {
+        return new Promise((resolve, reject) => {
+            userProfileService
+                .recoverAccount(user.value.hdid)
+                .then((userProfile) => {
+                    logger.debug(
+                        `recoverUserAccount: ${JSON.stringify(userProfile)}`
+                    );
+                    setProfileUserData(userProfile);
+                    resolve();
+                })
+                .catch((resultError: ResultError) => {
+                    setUserError(resultError.resultMessage);
+                    reject(resultError);
+                });
+        });
+    }
+
+    function retrieveEssentialData(): Promise<void> {
+        return new Promise((resolve) => {
+            status.value = LoadStatus.REQUESTED;
+            patientService
+                .getPatient(user.value.hdid)
+                .then((result: Patient) => {
+                    if (!result) {
+                        logger.debug("Patient retrieval failed");
+                        patientRetrievalFailed.value = true;
+                        resolve();
+                        return;
+                    }
+                    setPatient(result);
+                    userProfileService
+                        .getProfile(user.value.hdid)
+                        .then((userProfile) => {
+                            logger.debug(
+                                `User Profile: ${JSON.stringify(userProfile)}`
+                            );
+                            setProfileUserData(userProfile);
+                            resolve();
+                        })
+                        .catch((resultError: ResultError) => {
+                            logger.debug(
+                                `User Profile retrieval failed, status (${resultError.statusCode})`
+                            );
+                            handleError(
+                                resultError,
+                                ErrorType.Retrieve,
+                                ErrorSourceType.Profile
+                            );
+                            resolve();
+                        });
+                })
+                .catch((resultError: ResultError) => {
+                    logger.debug(
+                        `Patient retrieval failed, status (${resultError.statusCode})`
+                    );
+                    handleError(
+                        resultError,
+                        ErrorType.Retrieve,
+                        ErrorSourceType.Patient
+                    );
```

Comment:
```suggestion
                    if (error.statusCode === 429) {
                        logger.debug(
                            "Patient retrieval failed because of too many requests"
                        );
                        appStore.setAppError(AppErrorType.TooManyRequests);
                    } else {
                        logger.debug("Patient retrieval failed");
                        patientRetrievalFailed.value = true;
                    }
```
This should be dispatching the app-wide error or committing that patient retrieval failed.